### PR TITLE
Arm asm fix

### LINKED ISSUE #3846: Compiling on ARM 64bit
Hi all.

I'm building `Pioneer` on `ARM-64bit` architectures for Fedora. The release `20160907` is successfully compiled: [log](https://kojipkgs.fedoraproject.org//packages/pioneer/20160907/1.fc26/data/logs/aarch64/build.log) / [list software used](https://kojipkgs.fedoraproject.org//packages/pioneer/20160907/1.fc26/data/logs/aarch64/root.log)

Latest release `20161022` is failing on `ARM64` but i can't recognize the reason. Can you take a look please?
[Build log](https://kojipkgs.fedoraproject.org//work/tasks/8451/16188451/build.log)
[Dependencies log](https://kojipkgs.fedoraproject.org//work/tasks/8451/16188451/root.log)

Thanks.

### PR DESCRIPTION
**Pull request by fluffyfreak**
This fixes #3846 without random distro hacking.

### CODE REVIEW

File: contrib/profiler/Profiler.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #undef fastcall
 
 //#define USE_CHRONO
-#if !defined(USE_CHRONO) && defined(__arm__)
+#if !defined(USE_CHRONO) && defined(__arm__) || defined(__aarch64__)
```
Comment (senderghost) [inline contrib/profiler/Profiler.h]: This part is not correct, in my opinion. It should be:
`#if !defined(USE_CHRONO) && (defined(__arm__) || defined(__aarch64__))`

Comment (fluffyfreak) [inline contrib/profiler/Profiler.h]: that won't compile

Comment (senderghost) [inline contrib/profiler/Profiler.h]: > that won't compile

Test case for what you proposed:
```cpp
#include <stdio.h>

#if !defined(USE_CHRONO) && defined(__arm__) || defined(__aarch64__)
#define USE_CHRONO
#endif

int main ()
{
	printf("USE_CHRONO: %s\n",
#ifdef USE_CHRONO
	"defined"
#else
	"not defined"
#endif
	);

	return 0;
}
```
```
% cc test.c && ./a.out
USE_CHRONO: not defined

% cc -DUSE_CHRONO test.c && ./a.out
USE_CHRONO: defined

% cc -D__arm__ test.c && ./a.out
USE_CHRONO: defined

% cc -D__aarch64__ test.c && ./a.out
USE_CHRONO: defined

% cc -DUSE_CHRONO -D__arm__ test.c && ./a.out
USE_CHRONO: defined

% cc -DUSE_CHRONO -D__aarch64__ test.c && ./a.out
test.c:4:9: warning: 'USE_CHRONO' macro redefined
#define USE_CHRONO
        ^
<command line>:1:9: note: previous definition is here
#define USE_CHRONO 1
        ^
1 warning generated.
USE_CHRONO: defined
```

Test case for what I proposed:
```cpp
#include <stdio.h>

#if !defined(USE_CHRONO) && (defined(__arm__) || defined(__aarch64__))
#define USE_CHRONO
#endif

int main ()
{
	printf("USE_CHRONO: %s\n",
#ifdef USE_CHRONO
	"defined"
#else
	"not defined"
#endif
	);

	return 0;
}
```
```
% cc test2.c && ./a.out
USE_CHRONO: not defined

% cc -DUSE_CHRONO test2.c && ./a.out
USE_CHRONO: defined

% cc -D__arm__ test2.c && ./a.out
USE_CHRONO: defined

% cc -D__aarch64__ test2.c && ./a.out
USE_CHRONO: defined

% cc -DUSE_CHRONO -D__arm__ test2.c && ./a.out
USE_CHRONO: defined

% cc -DUSE_CHRONO -D__aarch64__ test2.c && ./a.out
USE_CHRONO: defined
```
Did you see where your part is not correct?

Comment (fluffyfreak) [inline contrib/profiler/Profiler.h]: yes, you're correct

**Comment by impaktor** [comment]
Were any of those commits fixes to previous? Then getting  `git add <stuff>; git commit --amend` into the finger tips when (force) pushing fixes makes for a tidy PR.

**Pull request merged**